packages/ui-library/src/utils/lit-element-custom.ts

```diff
@@ -0,0 +1,31 @@
+import { LitElement, PropertyDeclaration } from 'lit';
+
+export const camelCaseToKebabCase = (str: string) => str.replace(/([a-z0-9])([A-Z])/g, '$1-$2').toLowerCase();
```

Comment:
this regular expression might cause problems with some specific attribute names, e.g. `XMLValue` would be transformed into `xmlvalue` but should be `xml-value`.

In this stackOverflow article is a different solution the covers more special cases:
https://stackoverflow.com/a/67243723

The code is not that well readable though, here a version with some better naming:
```js
const kebabize = (str) => str.replace(/[A-Z]+(?![a-z])|[A-Z]/g, (upperCaseGroup, isNotFirstCharacter) => (isNotFirstCharacter ? "-" : "") + upperCaseGroup.toLowerCase())
```

I would still add a comment above this function to explain what it is doing, something like: "This regEx goes through the whole input string and searches for a consecutive group of upper case letters, a group can also consist of only one element. For each group found the callback function is called, if the group starts with the first character of the input string only make the group lower case else add a "-" at the front of the group and then make everything lower case"

Reply:
Thanks for the hint, now adjusted with the comment.